### DISCUSSION TITLE
A Response to Some Confusion About the New Look of Zettlr 2.0

### DISCUSSION THREAD

**Discussion by nathanlesage** [Zettlr/Zettlr#2606] [General]
Hey all,

I'm receiving a few confused comments as to why Zettlr 2.0 looks so different than 1.8.9 and people are demanding back the "old" 1.8.9 looks. I am starting this thread to explicate my reasoning behind the new looks of Zettlr and so that we can all discuss on the same grounds, because I am curious as to what might be causing it.

So, initially when I started developing Zettlr, the only experience I had was more or less with the apps I daily used (Mail software, browsers, PDF readers, Zotero, Word, etc.) and with web design. The reason why I chose Electron as the underlying framework was both because it's easy to get something up and running quickly across different platforms, because when I develop something I don't want to develop it solely for one platform, since I myself frequently switch between Windows and macOS and can't stand it if I have to use some worse alternatives for one of the platforms of the programs I often use.

But the drawback was always that Electron apps are sluggish and too big for what most of them do. So I wanted to reduce the drawbacks as much as possible while keeping the cross-platform benefits of the framework.

When I first developed the app, I was automatically developing a website, because if you begin to write HTML code you kind of automatically fall into the habits of writing anything as you would a website, which is completely understandable and the normal reaction. HTML is a web language, all websites are written in HTML, so when I write HTML I'm writing a website.

However, the thing with Electron is that it's *not* a website, but a desktop app. Yes, it's leveraging a browser to do the job, but that should not distract from the fact that it's still an executable you run even if you're offline. So when you start an app, it should *not* look and feel like a website. The previous design of Zettlr was partly because I didn't have the means to develop on multiple platforms and I didn't have that much experience, so a uniform design was the reasonable choice. However, even back then I knew I didn't want to keep that. Later, I was confirmed by many discussions I read online about how bad Electron apps can be if you do it wrong.

And think about it: No matter which Electron-based app you use – let it be Discord, Slack, Spotify (not really Electron but still web technologies), Teams – you name it –, all of them feel sluggish and somehow not right. Why should you install a 300MB app if you can just use the PWA? There is no real reason behind that.

So I knew I wanted to change that, and since most functions were already implemented in 1.8.9 I had the time to actually do what I really wanted to do and spent weeks tweaking the design so that it looks native on macOS and Windows, and more or less native on Linux. On Linux it's much more difficult, since there's a plethora of window managers, but at least one can try.

I oriented myself closely at all the available resources on native design (Apple HIG, Windows Metro Design Guidelines, GNOME HIG) and refrained from doing anything funky, unless I had to. The result is an app that leverages all available resources so that it looks extremely native, and with a lot of other changes to the code base, the app now feels snappy and if you don't look for it, you aren't going to notice that it's actually an Electron app. And that's what I wanted to achieve.

Even more reasons (and more elaborate than a short discussion starter on GitHub) are outlined in my two pieces on this specific matter; feel free to read through them [[1]](https://www.hendrik-erz.de/post/electron-bloated-web-and-trade-offs) [[2]](https://www.hendrik-erz.de/post/ux-design-is-not-for-creative-people)!

***

Some people might ask now why I start this discussion this early, given that these are just a few opinions, but I believe that addressing concerns early on and transparently, as well as delivering the reasoning behind design decisions is the way to go.

**Comment by Klemet** [comment]
Hey, Hendrik !

I wanted to thank you for this detailed explanation of your thinking behind this huge change in Zettlr's appearance. Your two articles that you add at the end explains a lot, and were really interesting to read !

However, while I think that I understand your line of thinking, from the point of view of the user that I am, I feel like it's a big loss. Zettlr < 2.0 was one one the most beautiful software I had on my computer. As an avid user of Joplin, I even customized Joplin's css to incorporate some of the designs choices that you made into it.

But now, on Windows...I can't help but not feeling very good when I open Zettlr. And I think that's understandable, as I went from this :

![Zettlr_interface](https://user-images.githubusercontent.com/41070328/137830902-1f844779-fc02-408b-8ad7-119fb35a00c7.png)

To this :

![image](https://user-images.githubusercontent.com/41070328/137831012-2185d1f7-ebc6-4c74-9cf0-6dd223767483.png)

So from a user perspective, I go from a "Flat", minimalist design that had great colours and a great character to something that looks a bit like, from my own sensitivity and memories, a software from the early 2000. It's pretty...grey-ish ? And I don't know why, but something feels wrong about the layout. My brain keeps telling me that something isn't right in the balance of the things that it is seeing, which I never felt with the previous versions.

There are even thick, black borders everywhere, which are very reminiscent of my first websites in HTML where I used to put them on every blocks, thinking that it would make things clearer.

![image](https://user-images.githubusercontent.com/41070328/137831562-39ba473a-d20c-450f-af73-b94beb65c338.png)


Yet knowing you, and according to what you wrote, I know that you must have done a lot of work to follow Microsoft's guidelines on design. But as a user, I never felt like I really wanted an app that looks like the other Microsoft apps; what I wanted was an app that makes me smile every time I open it, that made me want to use it, and that was easy to use, Microsoft be damned.

And on that last aspect, I'm not sure that it's a gain either; the icons of Zettlr are now much smaller, making it harder on my eyes to see at a glance what I should click on. I also now have a menu bar that I never really use, since it's hidden by default in most of the software that I use on Windows today, leading to more elements that my brain has to process.

So yeah, maybe this is a big victory against the bloated web and the bloated software made with electron. But as I user, I feel it like a loss. To cite the meme of Marie Kondo, _this does not sparkle joy_ anymore to me.

So as aesthetics are very important to me (because I look at my computer screen for the major part of the day), I now feel like looking elsewhere than Zettlr for my Markdown academics needs. And I'm really not saying that as a threat, or to try to punish you for your choice; I'm pretty sure that you're around 800% more knowledgeable than me in software development and design, and I'm really confident that Zettlr will go smoothly in the future with you at its helm. I'll still recommend it to my colleagues in research too ! 

I just wanted to give you my honest opinion on the design, and maybe try to convey what other users might feel - which I think might be the case, seeing your message. Those are just my feelings, and the design of Zettlr should not be tailored to them. I understand that it's impossible to satisfy every user, and I'm really at peace with this.

So I hope that you'll reward yourself on all of the good work done on v2.0, which is a big milestone 🎊 ! I also hope that you'll find a way to be at peace yourself with the fact that some users might be at odds with this new design; maybe it will grow on them. Maybe it'll grow on me ! In any case, the reasoning behind it is very interesting, and I'm very curious to know what others are thinking about it.

**Reply by nathanlesage** [reply]
Thank you for your warm words! You know what's curious? When I started implementing native styles it really was super easy doing that on macOS, because everything felt great, but on Windows, it was always a tad lacking. And the interesting insight I gained during this journey was that even Microsoft apparently hates its own design, since many Microsoft-internal apps, such as MS Teams, hell, even the Office suite, clearly violates their own design standards.

It was definitely hard for me, but I set myself this goal and I think I succeeded, albeit I'm totally with you that the Metro design is not that great. The thick black borders? Actually everywhere on Windows! The gray not necessarily.

But then I have to admit: I'm a Mac user, not a Windows user, so not as much work went into the Windows styling as went into the macOS styling. Throughout the development process I had hoped for people using Linux and Windows joining in, but that unfortunately didn't happen.

That being said: I understand if the current design doesn't suit your aesthetic senses, and it certainly doesn't suit mine, but I do hope that someone – even if not you – will take it on them and do a sweep over the Windows design and make it look better. Or, what's also possible, to re-integrate the old styles into the app as a new theme people can choose if they like! So before you leave Zettlr, consider that! I always encourage people to help make the app better! :)

**Reply by Klemet** [reply]
That's a very interesting insight into Microsoft's design guidelines ! Now that you've said it, I've started to notice those things in Window 10's menus. I think I understand why they didn't go that route and went for a "flatter" design on office !

I'm sorry to hear that it was hard for you, and sorry that there wasn't much effort helping you with the Windows styling. I'll put it on my list to maybe take a look at the CSS, and see if I can imitate the design of the office apps - which could be a bit better. But seeing as I'm swimming in work, I can't promise anything :c. Yet, I'd love it so that I can give back to Zettlr what it gave to me !

**Reply by nathanlesage** [reply]
> But seeing as I'm swimming in work, I can't promise anything

My go-to answer since a few weeks and until the end of the year :( So no hurry …

**Reply by Klemet** [reply]
As you're doing a PhD and making an awesome and very ambitious software on the side, I'd say that it's understandable : ). A wise person once said that productive people quickly realize that they can't do much in a year; but that they can accomplish mush in a decade. I'm sure it will come to pass !

**Reply by StoltHD** [reply]
> That's a very interesting insight into Microsoft's design guidelines ! Now that you've said it, I've started to notice those things in Window 10's menus. I think I understand why they didn't go that route and went for a "flatter" design on office !
> 
> I'm sorry to hear that it was hard for you, and sorry that there wasn't much effort helping you with the Windows styling. I'll put it on my list to maybe take a look at the CSS, and see if I can imitate the design of the office apps - which could be a bit better. But seeing as I'm swimming in work, I can't promise anything :c. Yet, I'd love it so that I can give back to Zettlr what it gave to me !

If you use the dark mode/night mode, it doesn't look that "bad", it is actually useable.
Personally I set the "Frankfurt" theme, set "Night Mode", and turn of the "Automatically turn on Night Mode", that way I get the Night mode permanently, and it is easier to my eyes... The menues, the icons, the black borders is not that prominent anymore:
![image](https://user-images.githubusercontent.com/28066567/137945815-bc5319bb-1444-445a-9707-925f83cbf18f.png)

But sadly because I need to create wiki-links inside md-tables, and want to have wiki-links with aliases, there are other problems for me, both with usage and display of data...

but I still use Zettlr as my default md editor because I actually like what I look at, and the GUI for me is not that big of an difference from version 1.8.x since I use the "Night Mode", and use Foam and Obsidian as "workspaces" for my research.

I really hope Zettlr will add support for wiki-links piped aliases, and escaped piped wiki-links with aliases in md-tables...
Then I would have a near complete set of tools, that each do some of the work I need...

---
but... I am a little biased maybe, because I have actually never been a huge fan of the visual side of Apples GUI.
